src/kinvey.d.ts

```diff
@@ -170,7 +170,7 @@ export namespace Kinvey {
 
   // DataStore class
   abstract class DataStore {
-    static collection<T extends Entity>(collection: string, type?: DataStoreType, options?: {
+    static collection<T extends Entity = Entity>(collection: string, type?: DataStoreType, options?: {
```

Comment:
Wouldn't this throw errors in compiler versions before 2.3?

What exactly is the benefit of having the default value the same as the base type value? Meaning if the default value was an extension of Entity, this would have some meaning, but having it the same... I can't tell what the benefit is and if this doesn't compile unless your tsconfig has an "ignore definition check" flag, it's not worth it. Isn't this the same behavior as before?

Reply:
This would throw errors before version 2.3.

Before it  required you to provide a type that was an `Entity`. The benefit is that you do not have to provide a type since the default is an `Entity`. The difference in code looks like this:

```javascript
// Before
const datastore = Kinvey.DataStore.collection('books', Kinvey.DataStoreType.Network) as Kinvey.NetworkStore<Kinvey.Entity>;

// After
const datastore = Kinvey.DataStore.collection('books', Kinvey.DataStoreType.Network) as Kinvey.NetworkStore;
```

Reply:
Hm... I think this is the effect of line [182](https://github.com/Kinvey/angular2-sdk/pull/6/files#diff-5f10b613d234f1a9aa800e78deb0b7d1R182).

Changing the `collection()` method shouldn't affect how you use the `NetworkStore` class to cast, should it? I think this particular amendment to `collection()` has no effect on anything, I believe. I think you can still omit the type parameter and it's treated as a `Kinvey.Entity`.

Reply:
In this case there is no way to be backwards compatible while fixing this issue. Currently without this changes someone will receive errors that prevent them from running their application.